internal/controller/postgrescluster/instance_test.go

```diff
@@ -1573,3 +1576,178 @@ func TestFindAvailableInstanceNames(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcileInstanceSetPodDisruptionBudget(t *testing.T) {
+	env, cc, config := setupTestEnv(t, ControllerName)
+	t.Cleanup(func() { teardownTestEnv(t, env) })
+
+	r := &Reconciler{}
+	ctx, cancel := setupManager(t, config, func(mgr manager.Manager) {
+		r = &Reconciler{
+			Client:   cc,
+			Owner:    client.FieldOwner(t.Name()),
+			Recorder: mgr.GetEventRecorderFor(ControllerName),
+			Tracer:   otel.Tracer(t.Name()),
+		}
+	})
+	t.Cleanup(func() { teardownManager(cancel, t) })
+
+	foundPDB := func(
+		t *testing.T,
+		cluster *v1beta1.PostgresCluster,
+		spec *v1beta1.PostgresInstanceSetSpec,
+		minAvailable int32,
+	) bool {
+		pdb, err := r.generatePodDisruptionBudget(
+			cluster,
+			naming.InstanceSetPodDisruptionBudget(cluster, spec),
+			minAvailable,
+			naming.ClusterInstanceSet(cluster.Name, spec.Name),
+		)
+		assert.NilError(t, err)
+		got := &policyv1beta1.PodDisruptionBudget{}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(pdb), got)
+		return !apierrors.IsNotFound(err)
+
+	}
+
+	ns := &corev1.Namespace{}
+	ns.GenerateName = "postgres-operator-test-"
+	ns.Labels = labels.Set{"postgres-operator-test": t.Name()}
+	assert.NilError(t, cc.Create(ctx, ns))
+	t.Cleanup(func() { assert.Check(t, cc.Delete(ctx, ns)) })
+
+	t.Run("empty", func(t *testing.T) {
+		cluster := &v1beta1.PostgresCluster{}
+		spec := &v1beta1.PostgresInstanceSetSpec{}
+
+		assert.Equal(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec).Error(),
+			"Replicas should be defined")
+	})
+
+	t.Run("not created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.Namespace = ns.Name
+		spec := &cluster.Spec.InstanceSets[0]
+		spec.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
+		assert.NilError(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec))
+		assert.Assert(t, !foundPDB(t, cluster, spec, 0))
+	})
+
+	t.Run("created", func(t *testing.T) {
+		cluster := testCluster()
+		cluster.UID = types.UID("hippouid")
+		cluster.Namespace = ns.Name
+		spec := &cluster.Spec.InstanceSets[0]
+		spec.MinAvailable = &intstr.IntOrString{IntVal: int32(1)}
+		assert.NilError(t, r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec))
+		assert.Assert(t, foundPDB(t, cluster, spec, 1))
+
+		t.Run("delete pdb", func(t *testing.T) {
+			spec.MinAvailable = &intstr.IntOrString{IntVal: int32(0)}
+			err := r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec)
+			if apierrors.IsConflict(err) {
+				err = r.reconcileInstanceSetPodDisruptionBudget(ctx, cluster, spec)
+			}
+			assert.NilError(t, err, errors.Unwrap(err))
+			assert.Assert(t, !foundPDB(t, cluster, spec, 0))
+		})
+	})
+}
+
+func TestDeleteOrphanedPodDisruptionBudgets(t *testing.T) {
+	env, cc, config := setupTestEnv(t, ControllerName)
+	t.Cleanup(func() { teardownTestEnv(t, env) })
+
+	r := &Reconciler{}
+	ctx, cancel := setupManager(t, config, func(mgr manager.Manager) {
```

Comment:
🔧 This test is calling reconcile methods directly, so there should be no Manager.